### ISSUE TITLE
`yarn config set` can't set `--<command>.<flag>` values

### ISSUE THREAD

**Issue by pixelastic** [yarnpkg/yarn#6499]
Hello yarn team :wave: ,

**Do you want to request a *feature* or report a *bug*?**
A bug :bug: 

**What is the current behavior?**
I'm trying to set `--run.silent true` to my `.yarnrc` through the use of `yarn config set` but the `--` seem to throw the command off.

**If the current behavior is a bug, please provide the steps to reproduce.**
I tried the following syntaxes, but they all fail with `error Invalid subcommand. Try "set, get, delete, list, current"`

- `yarn config set --run.silent true`
- `yarn config set "--run.silent" true`
- `yarn config set \-\-run.silent true`

**What is the expected behavior?**
[Documentation of `.yarnrc`](https://yarnpkg.com/lang/en/docs/yarnrc/) states that:
![image](https://user-images.githubusercontent.com/283419/46531092-41408180-c89c-11e8-9363-1244eeabcb59.png)

`.yarnrc` contains the comment `# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.`, making me think that the only way to set config is through `yarn config set`.

[Documentation of `yarn config`](https://yarnpkg.com/en/docs/cli/config) states that:
![image](https://user-images.githubusercontent.com/283419/46531112-4ef60700-c89c-11e8-8eee-b6b8b46f83f0.png)

Knowing all that, I would expect `yarn config set` to allow me to add `--`-prefixed values.

**Please mention your node.js, yarn and operating system version.**
- `node -v`: `v10.11.0`
- `yarn -v`: `1.12.0`
- Ubuntu 18.04

**Comment by arcanis** [comment]
> .yarnrc contains the comment # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY., making me think that the only way to set config is through yarn config set.

Yep, that's the actual bug. The yarnrc file is totally safe to modify manually as long as you now what you're doing.

This header is just there because the same serializer is used to generate the lockfile, which IS generated and shouldn't be modified, and I guess it was easier at the beginning to just print the warning in every case. We should improve this.

**Comment by pixelastic** [comment]
Ah thanks, that's good to know.

I was actually looking for a way to have `.yarnrc` as part of my dotfiles repo and symlink it to `~/.yarnrc` on any fresh install. As the file seemed auto-generated I tried `yarn config set` instead, but if it's not needed I'll do a simple link.

That being said, my file does contain a `lastUpdateCheck 1538717593702` line. I'm not sure what this is about, but my guess is that this value will evolve over time and if I symlink it will update my source file as well (which will be inconvenient).

I can live with having to manually set the value in the file for now, but I think solving the `yarn config set` parsing might be a better long-term solution.

Thanks for the clear and quick response, though
